### ISSUE TITLE
How to handle overfitting ? 

### ISSUE THREAD

**Issue by astranovasky** [facebookresearch/detr#342]
I trained DETR in a custom dataset. I applied data augmentation techniques by adding noise (small black squares in the image) to some of the input images. 
However, my model seems to overfit pretty quickly, already from the 7th epoch. 

Could I use any other regularization method or use more transformations during the data augmentation?
The images and their objects have almost no differences in rotation and scale, so i suppose that rotation and scaling data augmentation methods will not help. 

Below you can see the learning curve of the model. 

![image](https://user-images.githubusercontent.com/1334058/109510678-994a3500-7aa2-11eb-8e25-a95b160fb849.png)

**Comment by alcinos** [comment]
Hi @stamatelou 
As usual in Deep learning I'm afraid there is no silver bullet to combat the overfitting. Here are a few non-exclusive general directions you could try:

1. Reduce capacity: smaller backbone and/or smaller transformer (either depth or width). You could also try reducing the number of object queries.
2. Finetuning: If your dataset is small, you could start from a checkpoint trained on COCO, then finetune on your dataset with smaller learning rate
3. Regularization: Higher dropout, higher weight decay,...
4. Data augmentation: See for eg https://github.com/harpalsahota/bbaug

Also sometimes, the losses seem to indicate quite some overfitting, but in reality the performance (as measured with AP) is still increasing.
Best of luck
